Sources/MapboxNavigation/NavigationCameraStateTransition.swift

```diff
@@ -102,6 +107,16 @@ public class NavigationCameraStateTransition: CameraStateTransition {
     }
     
     public func updateForFollowing(_ cameraOptions: CameraOptions) {
+        // Check whether the original and target locations and headings are larger than a certain threshold when current camera state is following.
+        guard let mapView = mapView,
+              let center = cameraOptions.center,
+              let zoom = cameraOptions.zoom,
+              let bearing = cameraOptions.bearing else { return }
+
+        let centerUpdateThreshould = baseCenterUpdateThreshould * pow(2, 22.0 - Double(zoom))
+        guard mapView.cameraState.center.distance(to: center) > centerUpdateThreshould,
+              abs(mapView.cameraState.bearing - bearing) > bearingUpdateThreshold else { return }
+        
```

Comment:
The way this guard statement is currently expressed, it would ignore any animation request that doesn’t change _both_ the map’s center location and its bearing. But there are situations where only one of the two needs to be animated, since this method serves as a bottleneck for all animations. Instead of returning early, I think we’ll need to conditionalize each section of the code below on whether it would have an effect.

Reply:
I see. Thanks for the suggestion. I'll move it to specific check.